src/process_utility.c

```diff
@@ -2563,19 +2566,50 @@ process_altertable_validate_constraint_end(Hypertable *ht, AlterTableCmd *cmd)
 }
 
 static void
-process_altertable_drop_not_null(Hypertable *ht, AlterTableCmd *cmd)
+validate_uses_hypercore(Hypertable *ht, Oid chunk_relid, void *arg)
 {
-	int i;
+	AlterTableCmd *cmd = (AlterTableCmd *) arg;
+	Chunk *chunk = ts_chunk_get_by_relid(chunk_relid, true);
+	if (cmd->subtype == AT_SetNotNull)
+	{
+		if (ts_chunk_is_compressed(chunk) && !ts_is_hypercore_am(chunk->amoid))
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("operation not supported on compressed hypertables not using table "
+							"access method"),
```

Comment:
People will probably not understand this. What is "the table access method". We cannot assume that people know that we are referring to _our_ Hypercore table access method.

```suggestion
					 errmsg("operation not supported on compressed chunks not using the"
							"Hypercore table access method"),
```